adflow/mphys/mphys_adflow.py

```diff
@@ -178,7 +196,10 @@ def setAeroProblem(solver, ap, ap_vars, inputs=None, outputs=None, print_dict=Tr
 
         updatesMade = set_vol_coords(solver, inputs)
 
-    if outputs is not None:
+    # We need to make sure this isn't the first optimization iteration
+    # because OM will overwrite the ADflow state vector with all ones
+    # before the first iteration.
+    if outputs is not None and ap.adflowData.callCounter != -1:
         updatesMade = set_states(solver, outputs)
```

Comment:
I guess this is fine for now, but in future a better approach would be to find a way to actually set the initial value of the OpenMDAO outputs to the freestream initial conditions used by ADflow (can we do this during setup or configuration?). This approach of ignoring OpenMDAO inputs and doing stuff behind its back always gives me a bad feeling

Reply:
I agree I think we should be able to do this without going behind OM's back as well.  I will see if I can figure something out.

Reply:
Think it might not be possible to do it by directly specifying the values when calling `add_output` because that is called in the `setup` where we don't even know the state vector size yet right? Let alone the freestream conditions.
https://github.com/mdolab/adflow/blob/820f182d6ab2722ba92d0c41ea8c459d3a4095e6/adflow/mphys/mphys_adflow.py#L400

Could we do it using the `guess_nonlinear` method?

Reply:
We could potentially do it during guess nonlinear, however I think that is called everytime solve_nonlinear is called, so we would need logic to protect warm starts within coupled NLBGS sovler loops.

Reply:
I just removed the hacky logic we were using in `setAeroProblem` in the mphys layer to prevent OM from overwriting the initial states.  I will open a separate PR that addresses that issue without going behind OM's back.